src/core/auth.ts

```diff
@@ -214,14 +245,136 @@ export const auth = ({
     return result;
   };
 
+  /**
+   * Executing externalAuthenticationUrl mutation will prepare special URL which will redirect user to requested
+   * page after successfull authentication. After redirection state and code fields will be added to the URL.
+   *
+   * @param opts - Object withpluginId default value set as "mirumee.authentication.openidconnect" and input as
+   * JSON with redirectUrl - the URL where the user should be redirected after successful authentication.
+   * @returns Authentication data and errors
+   */
+  const getExternalAuthUrl: AuthSDK["getExternalAuthUrl"] = async opts => {
+    const result = await client.mutate<
+      ExternalAuthenticationUrlMutation,
+      ExternalAuthenticationUrlMutationVariables
+    >({
+      mutation: EXTERNAL_AUTHENTICATION_URL,
+      variables: { ...opts },
+    });
+
+    return result;
+  };
+
+  /**
+   * The externalObtainAccessTokens mutation will generate requested access tokens.
+   *
+   * @param opts - Object withpluginId default value set as "mirumee.authentication.openidconnect" and input as
+   * JSON with code - the authorization code received from the OAuth provider and state - the state value received
+   * from the OAuth provider
+   * @returns Login authentication data and errors
+   */
+  const getExternalAccessToken: AuthSDK["getExternalAccessToken"] = async opts => {
+    const result = await client.mutate<
+      ExternalObtainAccessTokensMutation,
+      ExternalObtainAccessTokensMutationVariables
+    >({
+      mutation: OBTAIN_EXTERNAL_ACCESS_TOKEN,
+      variables: { ...opts },
+    });
+
+    if (result.data?.externalObtainAccessTokens?.token) {
+      storage.setToken(result.data.externalObtainAccessTokens.token);
+    }
+
+    return result;
+  };
+
+  /**
+   * The mutation will prepare the logout URL. All values passed in field input will be added as GET parameters to the logout request.
+   *
+   * @param opts - Object withpluginId default value set as "mirumee.authentication.openidconnect" and input as
+   * JSON with returnTo - the URL where a user should be redirected
+   * @returns Logout data and errors
+   */
+  const logoutExternal: AuthSDK["logoutExternal"] = async opts => {
+    const result = await client.mutate<
+      ExternalLogoutMutation,
+      ExternalLogoutMutationVariables
+    >({
+      mutation: EXTERNAL_LOGOUT,
+      variables: { ...opts },
+    });
+
+    logout();
```

Comment:
It still waits for the response where it should be executed immediately. Move it above `client.mutate` call